homu/main.py

```diff
@@ -1027,6 +1027,15 @@ def create_merge(state, repo_cfg, branch, logger, git_cfg,
                     ok = False
             if state.squash:
                 try:
+                    first_commit = subprocess.check_output(
+                        git_cmd('cherry', 'origin/master',
```

Comment:
We ran into this again in clippy recently so it'd be nice to take another look at this (https://github.com/rust-lang/rust-clippy/pull/8356)

I believe this should be `merge_base_sha` rather than `origin/master`